### PR TITLE
Jv hw1

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,6 +20,15 @@
         </maven.checkstyle.plugin.configLocation>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+        </dependency>
```
Comment (ShemsheyMaxim) [inline pom.xml]: As for me, you can delete this dependency

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import com.example.bank.service.demo.entity.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User,Long> {
+    @Query(value = "from User u join fetch u.roles where u.id =?1")
+    Optional<User> findById(Long id);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: You can use default method JPA findById for this.

File: src/main/java/com/example/bank/service/demo/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    public void delete(User user) {
+        userRepository.delete(user);
+    }
+
+    @Override
+    public User getById(Long id) {
+        return userRepository.getOne(id);
```
Comment (ShemsheyMaxim) [inline src/main/java/com/example/bank/service/demo/service/impl/UserServiceImpl.java:31]: You can use findById(), it's default method which return Optional and in this method you can return User or throw exception

File: src/main/java/com/example/bank/service/demo/entity/Role.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import javax.persistence.Table;
+import lombok.Data;
+
+@Data
+@Entity
+@Table(name = "roles")
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
Comment (ShemsheyMaxim) [inline src/main/java/com/example/bank/service/demo/entity/Role.java]: Maybe need changed from long to Long

**Review by ShemsheyMaxim — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Sashk52) [reply pom.xml]: done

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: Thanks for advice. but i find this realization ok.

File: src/main/java/com/example/bank/service/demo/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (Sashk52) [reply src/main/java/com/example/bank/service/demo/service/impl/UserServiceImpl.java:31]: Thanks for advice. but i find this realization ok.

File: src/main/java/com/example/bank/service/demo/entity/Role.java (hunk 4 above)
Comment (Sashk52) [reply src/main/java/com/example/bank/service/demo/entity/Role.java]: done

**Review by ShemsheyMaxim — APPROVED** [review]

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java (hunk 2 above)
Comment (boroda4436) [reply src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: Agree with Max

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+    @Query(value = "from User u join fetch u.roles where u.phoneNumber =?1")
```
Comment (boroda4436) [inline src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: You don't need the `@Query` here

File: src/test/java/com/example/bank/service/demo/DemoApplicationTests.java [hunk 6]
```diff
@@ -2,12 +2,14 @@
...
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @SpringBootTest
 class DemoApplicationTests {
 
     @Test
     void contextLoads() {
+        assertTrue(true);
```
Comment (boroda4436) [inline src/test/java/com/example/bank/service/demo/DemoApplicationTests.java]: Do you need it here? The default implementation of this method is ok

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/bank/service/demo/DemoApplicationTests.java (hunk 6 above)
Comment (Sashk52) [reply src/test/java/com/example/bank/service/demo/DemoApplicationTests.java]: done

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: done

File: src/test/java/com/example/bank/service/demo/DemoApplicationTests.java (hunk 6 above)
Comment (Sashk52) [reply src/test/java/com/example/bank/service/demo/DemoApplicationTests.java]: done

File: src/main/java/com/example/bank/service/demo/repository/UserRepository.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/com/example/bank/service/demo/repository/UserRepository.java]: done

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**